[ADR-6] - secret-store-replacement.md (#4813)
cosmos/cosmos-sdk: docs/architecture/adr-006-secret-store-replacement.md

> # # ADR 006: Replace Keybase with platform specific secret store provided by the Keyring library

Reviewer: ```suggestion
# ADR 006: Replace Keybase with platform specific secret store provided by the Keyring library
```

## Changelog

> - July 29, 2019

Reviewer: ```suggestion
- July 29, 2019: Initial draft
```

## Context

> Currently, .gaiacli directory stores key material and metadata in a plain text database in the user’s home directory.  Key material is encrypted by a passphrase, protected by bcrypt hashing algorithm. Metadata (e.g. addresses, public keys, key storage details) is available in plain text. 

Reviewer: ```suggestion
Currently, an SDK application's CLI directory stores key material and metadata in a plain text database in the user’s home directory.  Key material is encrypted by a passphrase, protected by bcrypt hashing algorithm. Metadata (e.g. addresses, public keys, key storage details) is available in plain text. 
```
